xtask/src/examples.rs

```diff
@@ -143,16 +129,23 @@ impl ApplicationFunctions {
     }
 }
 
-trait OakExample {
+/// Construct run and build steps for some components of an `Example`.
+trait OakExampleSteps {
```

Comment:
That was my idea: the `OakExample` holds components to demonstrate functionality of Oak (e.g., for Trusted Shuffler). There is an `OakFunctionsExample`, too, which holds Oak Functions specific components. (I am really only renaming current functions).

The trait `OakExampleSteps` gives functions to run such an example.

WDYT?

Reply:
Yes I understand, I'm just not sure whether something like Trusted Shuffler should be considered an "example" and configured via TOML files etc. Personally I would just implement that logic directly in Rust, the whole point of xtask is to be able to use Rust instead of shell scripts. Anyways I think this is fine too, so please go ahead with it.